src/index.js

```diff
@@ -0,0 +1,245 @@
+exports.errors = errors
+
+/**
+ * Returns the version of latest migration.
+ * If no migrations are present returns 0.
+ *
+ * @param {array?} migrations - Array of migrations to consider. If undefined, the bundled migrations are used. Mainly for testing purpose.
+ * @returns {int}
+ */
+function getLatestMigrationVersion (migrations) {
+  migrations = migrations || defaultMigrations
+
+  if (!Array.isArray(migrations) || migrations.length === 0) {
+    return 0
+  }
+
+  return migrations[migrations.length - 1].version
+}
+
+exports.getLatestMigrationVersion = getLatestMigrationVersion
+
+/**
+ * Main function to execute forward migrations.
+ * It acquire lock on the provided path before doing any migrations.
+ *
+ * Signature of the progress callback is: function(migrationObject: object, currentMigrationNumber: int, totalMigrationsCount: int)
+ *
+ * @param {string} path - Path to initialized (!) JS-IPFS repo
+ * @param {Object} options - Options for migration
+ * @param {int?} options.toVersion - Version to which the repo should be migrated, if undefined repo will be migrated to the latest version.
+ * @param {boolean?} options.ignoreLock - Won't lock the repo for applying the migrations. Use with caution.
+ * @param {object?} options.repoOptions - Options that are passed to migrations, that can use them to correctly construct datastore. Options are same like for IPFSRepo.
+ * @param {function?} options.onProgress - Callback which will be called after each executed migration to report progress
+ * @param {boolean?} options.isDryRun - Allows to simulate the execution of the migrations without any effect.
+ * @param {array?} options.migrations - Array of migrations to migrate. If undefined, the bundled migrations are used. Mainly for testing purpose.
+ * @returns {Promise<void>}
+ */
+async function migrate (path, { toVersion, ignoreLock = false, repoOptions, onProgress, isDryRun = false, migrations }) {
+  migrations = migrations || defaultMigrations
+
+  if (!path) {
+    throw new errors.RequiredParameterError('Path argument is required!')
+  }
+
+  if (!(await repoInit.isRepoInitialized(path))) {
+    throw new errors.NotInitializedRepoError(`Repo in path ${path} is not initialized!`)
+  }
+
+  if (toVersion && (!Number.isInteger(toVersion) || toVersion <= 0)) {
+    throw new errors.InvalidValueError('Version has to be positive integer!')
+  }
+  toVersion = toVersion || getLatestMigrationVersion(migrations)
+
+  if (toVersion > getLatestMigrationVersion(migrations)) {
+    throw new errors.InvalidValueError('The ipfs-repo-migrations package does not have migration for version: ' + toVersion)
+  }
+
+  const currentVersion = await repoVersion.getVersion(path)
+
+  if (currentVersion === toVersion) {
+    log('Nothing to migrate, skipping migrations.')
+    return
+  }
+
+  if (currentVersion > toVersion) {
+    log(`Current repo's version (${currentVersion}) is higher then toVersion (${toVersion}), nothing to migrate.`)
```

Comment:
Should be an error IMHO since it will not leave the repo in the state the caller expects.